build-system/tasks/runtime-test.js

```diff
@@ -18,27 +18,38 @@ var argv = require('minimist')(process.argv.slice(2));
 var gulp = require('gulp-help')(require('gulp'));
 var Karma = require('karma').Server;
 var config = require('../config');
-var karmaConfig = config.karma;
-var extend = require('util')._extend;
 var fs = require('fs');
 var path = require('path');
 var util = require('gulp-util');
 var webserver = require('gulp-webserver');
 var app = require('../test-server').app;
 
+var karmaDefault = {
+  configFile: path.resolve('build-system/tasks/karma.conf.js'),
+  // "client" has to be specified here but not in configFile,
+  // because we want to change its value based on flags.
+  // The whole "client" object will get overridden if specified in config file.
+  client: {
```

Comment:
..... That's really weird.  And good catch! I think it's a bug before! I would suggest adding a comment in configFile saying that any object that will be modified later needs to be specified later in `runtime-test.js` to prevent future bug like this.